test/gtest/ucp/test_ucp_peer_failure.cc

```diff
@@ -488,6 +488,19 @@ class test_ucp_peer_failure_keepalive : public test_ucp_peer_failure
         add_variant_with_value(variants, UCP_FEATURE_AM | UCP_FEATURE_WAKEUP,
                                TEST_AM | WAKEUP, "am_wakeup");
     }
+
+    void wakeup_drain(const std::vector<entity*> &entities)
```

Comment:
so it's not just "drain" ; it actually is waiting for "deadline" time to make sure no more events arrive?
the name is a bit misleading..

Reply:
I guess that the name `drain` is suitable here - we're trying to read all wakeup events while 'deadline' timer isn't expired.

Reply:
maybe something like "wakeup_check_no_events"